library/core/src/str/pattern.rs

```diff
@@ -348,6 +348,45 @@ pub trait DoubleEndedSearcher<'a>: ReverseSearcher<'a> {}
 // Impl for char
 /////////////////////////////////////////////////////////////////////////////
 
+#[derive(Clone, Copy, Debug)]
+enum Utf8Size {
+    // Values are indexes, so `- 1`
+    One = 0,
+    Two = 1,
+    Three = 2,
+    Four = 3,
+}
```

Comment:
If wanted, I can change this to use a `rustc_scalar_valid_range` attribute, but it seems safer to use the workaround that everyone else in the ecosystem has to perform.

Reply:
Another option would be `One = 1` and so on, skipping 0. I don't know if that makes this clearer or less clear.